AWSClientRuntime/Sources/Signing/AWSSigV4Signer.swift

```diff
@@ -0,0 +1,68 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import ClientRuntime
+import AwsCommonRuntimeKit
+
+public class AWSSigV4Signer {
+    static let logger: SwiftLogger = SwiftLogger(label: "AWSSigV4Signer")
+    
+    public static func sigV4SignedURL(requestBuilder: SdkHttpRequestBuilder,
+                                      credentialsProvider: CredentialsProvider,
+                                      signingName: Swift.String,
+                                      signingRegion: Swift.String,
+                                      date: ClientRuntime.Date,
+                                      expiration: Int64) -> ClientRuntime.URL? {
```

Comment:
Question. Since this is the convenience function does it make sense to ever default the expiration or no?

Reply:
Good question -- we could have one, but then what would the default be?  5 minutes? 10 minutes? 1 hour? 1 day?  I don't have a good answer for this.

Also, it seems that some services probably have a maximum for how long a presigned url is valid (or there is something incredibly wrong in our code).  For example, when creating a presigned url w/ polly, I can seemingly only generate presigned urls with a maximum of 5 minutes, despite passing in a higher value like 86400 (24 hours).

Reply:
ya this is interesting, I guess for now we dont have one until we ever figure out what it should be or we get customer requests. Agree with your analysis here, thanks for providing it.